### PR TITLE
refactor(regex): use map->filter for simplicity

### PR DESCRIPTION
**Pull request by ajoslin**
I was reading through your code and had fun with this file. Ignore if
you wish.

### CODE REVIEW

**Comment by ajoslin** [comment]
Cool stuff, btw. You went ham on this.

Conventional changelog is gonna be the best thing on the internet.

**Comment by stevemao** [comment]
:+1: :smile:

File: lib/regex.js [hunk 1]
```diff
@@ -1,34 +1,22 @@
 'use strict';
-var forEach = require('lodash').forEach;
 
+function join(array, joiner) {
+  return array
+    .map(function(val) {
+      return (val || '').trim();
```
Comment (stevemao) [inline lib/regex.js:6]: Not sure if we need `|| ''` here.

Comment (stevemao) [inline lib/regex.js:6]: oh because of https://github.com/stevemao/conventional-commits-parser/blob/master/test/regex.spec.js#L25

Comment (ajoslin) [inline lib/regex.js:6]: You had an `if (val)` in the old version, so I added it just incase you were checking against the user providing null or something.

Comment (stevemao) [inline lib/regex.js:6]: I don't think it's needed. added in the doc saying this is `array` of `string`

**Pull request closed without merging**